velox/exec/tests/TableWriteTest.cpp

```diff
@@ -2152,6 +2156,251 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
+      groupingKeys,
+      std::vector<core::FieldAccessTypedExprPtr>{},
+      aggregateNames,
+      aggregates,
+      false, // ignoreNullKeys
+      std::move(tableWriterPlan.planNode()));
+
+  auto finalPlan =
+      tableWriterPlan.capturePlanNodeId(tableWriteNodeId_)
+          .localPartition(std::vector<std::string>{})
+          .tableWriteMerge(
+              std::move(writerOutputType), std::move(mergeAggregationNode))
+          .planNode();
+
+  auto result = AssertQueryBuilder(finalPlan).copyResults(pool());
+  auto rowVector = result->childAt(0)->asFlatVector<int64_t>();
+  auto fragmentVector = result->childAt(1)->asFlatVector<StringView>();
+  auto commitContextVector = result->childAt(2)->asFlatVector<StringView>();
+  auto columnStatsVector =
+      result->childAt(3 + partitionedBy_.size())->asFlatVector<int64_t>();
+
+  const int64_t expectedRows = 10 * 100;
+  std::vector<std::string> writeFiles;
+  int64_t numRows{0};
+
+  // For partitioned, expected result is as follows:
+  // Row     Fragment           Context       partition           c1_min_value
+  // null    null                x            partition1          0
+  // null    null                x            partition2          10
+  // null    null                x            partition3          15
+  // count   null                x            null                null
+  // null    partition1_update   x            null                null
+  // null    partition1_update   x            null                null
+  // null    partition2_update   x            null                null
+  // null    partition2_update   x            null                null
+  // null    partition3_update   x            null                null
+  //
+  // Note that we can have multiple same partition_update, they're for different
+  // files, but for stats, we would only have one record for each partition
+  //
+  // For unpartitioned, expected result is:
+  // Row     Fragment           Context       partition           c1_min_value
+  // null    null                x                                0
+  // count   null                x            null                null
+  // null    update              x            null                null
+
+  int statsRow = 0;
+  while (columnStatsVector->isNullAt(statsRow) && statsRow < result->size()) {
+    statsRow++;
+  }
+  for (int i = 1; i < result->size(); ++i) {
+    if (i < statsRow) {
+      ASSERT_TRUE(rowVector->isNullAt(i));
+      ASSERT_FALSE(fragmentVector->isNullAt(i));
+      ASSERT_TRUE(columnStatsVector->isNullAt(i));
+    } else if (i < result->size() - 1) {
+      ASSERT_TRUE(rowVector->isNullAt(i));
+      ASSERT_TRUE(fragmentVector->isNullAt(i));
+      ASSERT_FALSE(columnStatsVector->isNullAt(i));
+    } else {
+      ASSERT_FALSE(rowVector->isNullAt(i));
+      ASSERT_TRUE(fragmentVector->isNullAt(i));
+      ASSERT_TRUE(columnStatsVector->isNullAt(i));
+    }
+  }
+}
+
```

Comment:
Can you add tests in followup to verify the each type of stats are correctly generated? Thanks!